src/ui/text-color.ts

```diff
@@ -0,0 +1,153 @@
+import { settings } from "#app/system/settings/settings-manager";
+import { ShadowColor, TextColor, Color } from "#enums/color";
+import { TextStyle } from "#enums/text-style";
+import { UiTheme } from "#enums/ui-theme";
+
+/**
+ * Combination of a {@linkcode Color} and {@linkcode ShadowColor} used for text elements.
+ */
+interface TextColorCombination {
+  mainColor: Color;
+  shadowColor: ShadowColor;
+}
+
+/**
+ * Object linking each {@linkcode TextColor} to a {@linkcode Color} and {@linkcode ShadowColor}.
+ */
+const colorCombinations: { [key: string]: TextColorCombination } = {
```

Comment:
NIT: object should be frozen via `Object.freeze` to make it readonly (non manipulateable)